cloud/amazon/ec2_vol.py

```diff
@@ -332,7 +338,10 @@ def create_volume(module, ec2, zone):
     if volume is None:
         try:
             if boto_supports_volume_encryption():
-                volume = ec2.create_volume(volume_size, zone, snapshot, volume_type, iops, encrypted)
+                if kms_key_id:
```

Comment:
Point taken. Added the according check, actually merged the check from https://github.com/ansible/ansible-modules-core/pull/3064 and corrected the naming of the method.